From e204c887357f2d2ee1df5436a7d7f68b227c4b64 Mon Sep 17 00:00:00 2001
From: "plai@redhat.com" <plai@redhat.com>
Date: Fri, 22 Mar 2019 17:45:15 +0000
Subject: [PATCH 2/2] rhel: Set host-phys-bits-limit=48 on rhel machine-types

RH-Author: plai@redhat.com
Message-id: <1553276715-26278-3-git-send-email-plai@redhat.com>
Patchwork-id: 85126
O-Subject: [RHEL8.0 qemu-kvm PATCH 2/2] rhel: Set host-phys-bits-limit=48 on rhel machine-types
Bugzilla: 1688915
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>

From: Eduardo Habkost <ehabkost@redhat.com>

Upstream status: not applicable

Currently we use the host physical address size by default on
VMs.  This was a good default on most cases, but this is not the
case on host CPUs supporting 5-level EPT.  On those cases, we
want VMs to use 4-level EPT by default.

Ensure VMs will use 4-level EPT by default, by limiting physical
address bits to 48.

Not applicable upstream because upstream doesn't set
host-phys-bits=on by default.

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
(cherry picked from commit 01a2ecb4c38fe4a35455ea706e76984ee8d5a769)
Signed-off-by: Paul Lai <plai@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 include/hw/i386/pc.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 782d728..de25407 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -992,6 +992,11 @@ extern void igd_passthrough_isa_bridge_create(PCIBus *bus, uint16_t gpu_dev_id);
             .property = "host-phys-bits",\
             .value = "on",\
         },\
+        { /* PC_RHEL_COMPAT */ \
+            .driver = TYPE_X86_CPU,\
+            .property = "host-phys-bits-limit",\
+            .value = "48",\
+        },\
         { /* PC_RHEL_COMPAT bz 1508330 */ \
             .driver = "vfio-pci",\
             .property = "x-no-geforce-quirks",\
-- 
1.8.3.1

